File: src/main/java/com/example/hexagonal/port/dto/ProductDto.java
package com.example.hexagonal.port.dto;

import lombok.Builder;
import lombok.Data;

@Data
public class ProductDto {

  Long id;
  String name;
  Integer stock;

  public ProductDto() {
  }

  @Builder
  public ProductDto(Long id, String name, Integer stock) {
    this.id = id;
    this.name = name;
    this.stock = stock;
  }
}

File: kotlin/src/main/kotlin/com/example/hexagonal/application/port/dto/ProductDto.kt
package com.example.hexagonal.application.port.dto

import com.example.hexagonal.application.domain.Product.Companion.DEFAULT_NAME
import com.example.hexagonal.application.domain.Product.Companion.DEFAULT_STOCK

data class ProductDto(
    val id: Long,
    var name: String,
    var stock: Int? = 0,
) {
    companion object {
        fun aProductDtoOf(productId: Long) =
            ProductDto(id = productId, name = DEFAULT_NAME, stock = DEFAULT_STOCK)
    }
}
